lib/src/components/board/board.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:viam_sdk/src/resource/base.dart';
+
+import '../../gen/common/v1/common.pb.dart' as common;
+import '../../gen/component/board/v1/board.pbenum.dart';
+import '../../robot/client.dart';
+
+class BoardStatus {
+  final Map<String, int> analogs;
+  final Map<String, int> digitalInterrupts;
+
+  const BoardStatus(this.analogs, this.digitalInterrupts);
+
+  factory BoardStatus.fromProto(common.BoardStatus pbBoardStatus) {
+    BoardStatus boardStatus = BoardStatus(Map<String, int>(), Map<String, int>());
+    pbBoardStatus.analogs.forEach((key, value) => boardStatus.analogs[key] = value.value);
+    pbBoardStatus.digitalInterrupts.forEach((key, value) => boardStatus.digitalInterrupts[key] = (value.value as int));
+    return boardStatus;
+  }
+}
+
+/// Board represents a physical general purpose compute board that contains various
+/// components such as analog readers, and digital interrupts.
+///
+/// This acts as an abstract base class for any drivers representing specific
+/// board implementations. This cannot be used on its own.
+abstract class Board extends Resource {
+  static const Subtype subtype = Subtype(ResourceNamespaceRDK, ResourceTypeComponent, 'board');
+
+  @override
+  String name;
+
+  Board(this.name);
+
+  /// Get the status of the Board.
+  Future<BoardStatus> status({Map<String, dynamic>? extra});
+
+  /// Set the high/low state of the given pin of a board.
+  Future<void> setGpioState(String pin, bool high, {Map<String, dynamic>? extra});
+
+  /// Get the high/low state of the given pin of a board.
+  Future<bool> isGpioHigh(String pin, {Map<String, dynamic>? extra});
```

Comment:
still iffy on this name. trying to follow https://dart.dev/guides/language/effective-dart/design#prefer-a-non-imperative-verb-phrase-for-a-boolean-property-or-variable

Reply:
ok I've decided i don't like this. reverting back to `gpio`